### PR TITLE
ARTEMIS-805 old JMS clients failing on new broker

### PR DESCRIPTION
**Pull request by jbertram**
Old JMS clients are getting a CCE related to SessionBindingQueryResponseMessage

### CODE REVIEW

**Comment by clebertsuconic** [comment]
@jbertram  the failures showing at this build are real. I replicated the same issues through a local build.

I would run the whole testsuite for such changes.

**Comment by clebertsuconic** [comment]
this commit would actually break stuff pretty badly.

Even .artemis-distribution/src/test/scripts/validate-spaces.sh would be broken.

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java [hunk 1]
```diff
@@ -290,9 +292,19 @@ public int getServerVersion() {
 
    @Override
    public ClientSession.AddressQuery addressQuery(final SimpleString address) throws ActiveMQException {
-      SessionBindingQueryResponseMessage_V3 response = (SessionBindingQueryResponseMessage_V3) sessionChannel.sendBlocking(new SessionBindingQueryMessage(address), PacketImpl.SESS_BINDINGQUERY_RESP_V3);
-
-      return new AddressQueryImpl(response.isExists(), response.getQueueNames(), response.isAutoCreateJmsQueues(), response.isAutoCreateJmsTopics());
+      if (sessionChannel.supports(PacketImpl.SESS_BINDINGQUERY_RESP_V3)) {
```
Comment (TomasHofman) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java]: ```sessionChannel.supports()``` doesn't give correct results, I think I was trying that. The problem was that ```connection.getClientVersion()``` returned zero in ```ChannelImpl.supports()```, hence I was comparing version numbers directly.

Comment (TomasHofman) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java]: I checked and failing tests are passing on my local machine when using original commit.

**Comment by TomasHofman** [comment]
@clebertsuconic will this get into 1.4 branch?

**Comment by jbertram** [comment]
@TomasHofman, right now there is no 1.4 branch.

**Pull request closed without merging**